src/Microsoft.Data.SqlClient/src/Microsoft/Data/SqlClient/SqlConnectionString.cs

```diff
@@ -400,15 +406,15 @@ internal SqlConnectionString(string connectionString) : base(connectionString, G
                 }
             }
 
-            if (!_encrypt)
+            if (_encrypt == SqlConnectionEncryptOption.Optional)
             {    // Support legacy registry encryption settings
                 const string folder = "Software\\Microsoft\\MSSQLServer\\Client\\SuperSocketNetLib";
                 const string value = "Encrypt";
 
                 object obj = ADP.LocalMachineRegistryValue(folder, value);
                 if ((obj is int iObj) && (iObj == 1))
                 {         // If the registry key exists
-                    _encrypt = true;
+                    _encrypt = SqlConnectionEncryptOption.Mandatory;
```

Comment:
Shouldn't we support the `Strict` from the registry?

Reply:
I thought about it. I can't even find documentation for the way the setting currently behaves in netfx. I'm dis-inclined to extend it further since we will need something new in the future for system-level-forced-strict-encryption that is cross-plat.